crates/vhost-device-gpio/src/main.rs

```diff
@@ -9,7 +9,7 @@
 mod backend;
 #[cfg(target_env = "gnu")]
 mod gpio;
-#[cfg(test)]
+#[cfg(all(target_env = "gnu", any(test, feature = "mock_gpio")))]
```

Comment:
Any reason we need to depend on "gnu" here?

Reply:
Not sure if it is the same issue, but we had issues with Musl builds earlier for libgpiod support.

Reply:
yeah it was musl that tripped up gpio

Reply:
ACK. Not a topic of this this PR, but if we cannot meaningfully build with musl, we should maybe just avoid doing that instead of putting checks that disable almost everything across the code.

Reply:
You mean not build vhost-device-gpio at all?

Reply:
Yes. Not sure if it is possible through conditional workspace inclusions or needs to be done manually when invoking CI...

Reply:
I'm closing this for this PR as we are just following the pattern for the other bits of gpio.